packages/terra-form/src/components/ChoiceField.jsx

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable jsx-a11y/label-has-for */
+
+import React, { PropTypes } from 'react';
+
+import Control from './Control';
+import Field from './Field';
+
+const propTypes = {
+  /**
+   * Choices to populate radio buttosn for
+   */
+  choices: PropTypes.array,
+  /**
+   * The property name for the choice object used to set the choice label
+   */
+  choiceName: PropTypes.string,
+  /**
+   * The property name for the choice object used to set the choice value
+   */
+  choiceValue: PropTypes.string,
+  /**
+   * Error message for when the input is invalid
+   */
+  error: PropTypes.node,
+  /**
+   *  Help element to display with the input
+   */
+  help: PropTypes.node,
+  /**
+   * Whether the field is inline
+   */
+  isInline: PropTypes.bool,
+  /**
+   * Label of the input
+   */
+  label: PropTypes.node,
+  /**
+   * Name of the input attribute
+   */
+  name: PropTypes.string,
+  /**
+   * Function to trigger when user selects one of the radio buttons
+   */
+  onChange: PropTypes.func,
+  /**
+   * Initial Value of the input
+   */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  choices: [],
+  choiceName: 'name',
+  choiceValue: 'value',
```

Comment:
I don't know if that's better though. I see a need to pass in additional attributes to the individual nodes as well. I think we should just require the same api structure as that of the actual choice input, meaning we just rely on the name and value attributes straight up. Otherwise, this api will be very confusing for users to consume.